gamify-impl/src/main/java/ch/heigvd/gamify/domain/aggregate/RankingRepository.java

```diff
@@ -20,15 +20,23 @@
           + "HAVING SUM(r.points) > b.pointsLower "
           + "AND SUM(r.points) < b.pointsUpper "
           + "ORDER BY total DESC ")
-*/
 
-  @Query("SELECT r.category.idCategory.name AS category, e.user AS user, SUM(r.points) AS total "
-      + "FROM Rule r "
-      + "INNER JOIN Event e ON r.eventType = e.type AND r.id.app = e.app.name "
-      + "WHERE e.app.name = :app "
-      + "AND e.user = :userId "
-      + "AND r.category.idCategory.name = :category "
-      + "GROUP BY r.category.idCategory.name, e.user "
-      + "ORDER BY total DESC ")
+@Query(nativeQuery = true, value =
+        "SELECT p.points AS total, p.idx_user_user_id AS userId, ROW_NUMBER() OVER (ORDER BY p.points DESC) as rank "
+      + "FROM end_user_points p "
+      + "WHERE p.idx_user_app_name = :app "
+      + "AND p.idx_category_name = :category "
+      + "GROUP BY p.idx_user_user_id, p.points "
+      + "LIMIT :pagesize OFFSET :offset")
+
+ */
+  @Query(nativeQuery = true, value =
+          "SELECT p.points AS total "
+        + "FROM end_user_points p "
+        + "WHERE p.idx_user_app_name = :app "
+        + "AND p.idx_user_user_id = :userId "
+        + "AND p.idx_category_name = :category "
+        + "GROUP BY p.idx_user_user_id, p.points "
+        + "LIMIT 1")
```

Comment:
`LIMIT 1` ?

Reply:
This is a because we always get 1 user for 1 category. This should never filter out anything, but seems clearer on what it does. It can definitely be removed.

Reply:
Maybe we can check if we can return an `Optional` instead of a `List`. This will remove the need for the `LIMIT 1` as it would be clear for the user that we can only have one value.

If the `Optional` variant isn't available, we should discuss this again.